tests/commands/test_get.py

```diff
@@ -0,0 +1,465 @@
+    (['num'], ['num-3', 'num-16', 'num-20']),
+    (['unset', 'type'], ['a2cd', 'a13bc', 'a36ab']),
+    ([], ['a2cd', 'a13bc', 'a36ab'])])
+def test_get_sort(
+        repo: Repo, sort: Union[str, None], keys: list[str],
+        expected: list[str], default: Union[int, None]) -> None:
+    """
+    Test that `onyo get --keys x y z` with `-s` (ascending) or `-S`
+    (descending)  retrieves assets in the expected 'natural sorted' order.
+    """
+    cmd = ['onyo', 'get', '--keys', *keys, '-H']
+    cmd = cmd + [sort] if sort else cmd
+    ret = subprocess.run(cmd, capture_output=True, text=True)
+    output = [output.split('\t') for output in ret.stdout.split('\n')][:-1]
+
+    for i, key in enumerate(keys or ['type']):
+        if key == 'unset':  # nothing to be sorted
+            continue
+
+        assert [line[i if sort else -1] for line in output] == default if \
+            default else list(reversed(expected)) if sort == '-S' else expected
+
+    assert not ret.stderr
+    assert ret.returncode == 0
+
+
+def test_get_sort_error(repo: Repo) -> None:
+    """
+    Test that when using -s and -S simultaneously the appropriate error is
+    returned.
+    """
+    cmd = ['onyo', 'get', '-s', '-S']
+    ret = subprocess.run(cmd, capture_output=True, text=True)
+    msg = (
+        '--sort-ascending (-s) and --sort-descending (-S) cannot be used '
+        'together')
+    assert msg in ret.stderr
+    assert ret.returncode == 1
+
+
+@pytest.mark.parametrize('assets', [[
+    (Path('a13bc_foo_bar.1'), {'num': 'num-20', 'str': 'abc', 'id': '1'}),
+    (Path('a2cd_foo_bar.2'), {'num': 'num-3', 'str': 'def', 'id': '2'}),
+    (Path('a36ab_foo_bar.3'), {'num': 'num-16', 'str': 'ghi', 'id': '3'})]])
+@pytest.mark.parametrize('keys', [None, ['num'], ['str', 'num']])
+@pytest.mark.parametrize('reverse', [True, False])
+def test_natural_sort(
+        assets: list[tuple[Path, dict[str, str]]], keys: Union[list, None],
+        reverse: bool) -> None:
+    """Test implementation of natural sorting algorithm"""
+    sorted_assets = natural_sort(assets, keys=keys, reverse=reverse)
+    ids = [data.get('id') for _, data in sorted_assets]
+
+    if reverse:
+        ids = list(reversed(ids))
+
+    if keys is None:
+        assert ids == ['2', '1', '3']
+    elif keys[0] == 'num':
+        assert ids == ['2', '3', '1']
+    elif keys[0] == 'str':
+        assert ids == ['1', '2', '3']
+
+
+@pytest.mark.parametrize('assets', [[
+    (Path('a13bc_foo_bar.1'), {'num': 'num-20', 'str': 'abc'}),
+    (Path('a2cd_foo_bar.2'), {'num': 'num-3'}),
+    (Path('a36ab_foo_bar.3'), {'str': 'ghi'})]])
+@pytest.mark.parametrize('keys', [[
+    'type', 'make', 'model', 'serial', 'num', 'str', 'id']])
+def test_fill_unset(
+        assets: list[tuple[Path, dict[str, str]]], keys: list[str]) -> None:
+    """
+    Test that the `fill_unset()` function fills unset keys with the value
+    `'<unset>'`
+    """
+    unset_value = '<unset>'
```

Comment:
We've decided upon using `<unset>` for now. A follow-up PR will define a document with defaults that can be overwritten by the configuration. Addressing that flexibility here is out of scope for the PR.

Reply:
I feel like testing that this works and puts through the variable independently from the implementation, seems not that big of a change, considering that this was at many places hardcoded and sometimes missed during the multiple times it was refactored :) 

Can't this be simply done with one line of parameterization that passed `unset` into the function instead?

Reply:
Currently the value of `unset` is static, but there is a plan to make it dynamic. In that case, `unset_value` would simply be loaded from a submodule that either retrieves a custom unset value from the config, or `<unset>` if there is none. 

If I change the code to make it dynamic but then the way it is made dynamic changes (because `Filter` should not have an argument for unset values, this should be a default or configurable), I'm putting in extra work for something that has no relevance.

Furthermore, testing on whether unset is dynamic, and which values should be allowed, should be reserved for a PR focusing on the defaults.